spec/requests/api/stateless/members_spec.rb

```diff
@@ -23,13 +20,98 @@ def auth_headers
       get "/api/stateless/members/#{member.id}", nil, auth_headers
       expect(response.status).to eq(200)
-      expect(json_hash.keys).to include('id', 'first_name', 'last_name', 'email')
+      expect(json_hash.keys).to include('id', 'first_name', 'last_name', 'email', 'country', 'city', 'postal', 'address1', 'address2')
       expect(json_hash).to match({
-                                   id: member.id,
-                                   first_name: member.first_name,
-                                   last_name: member.last_name,
-                                   email: member.email
-                                 }.as_json)
+         id: member.id,
+         first_name: member.first_name,
+         last_name: member.last_name,
+         email: member.email,
+         country: member.country,
+         city: member.city,
+         postal: member.postal,
+         address1: member.address1,
+         address2: member.address2
+       }.as_json)
+    end
+  end
+
+  describe 'PUT update' do
+    context 'with valid parameters' do
+
+      let(:params) {{
+        member: {
+          first_name: "Harry",
+          last_name: "Tubman",
+          email: "test+1@example.com",
+          country: "United Kingdom",
+          city: "London",
+          postal: "12345",
+          address1: "Jam Factory 123"
+        }
+      }}
+
+      subject do
+        put "/api/stateless/members/#{member.id}", params, auth_headers
+      end
+
+      it 'updates the member locally and sends it back as json' do
+        subject
+        expect(json_hash).to match({
+                                     id: member.id,
+                                     first_name:  params[:member][:first_name],
+                                     last_name:  params[:member][:last_name],
+                                     email:  params[:member][:email],
+                                     country:  params[:member][:country],
+                                     city:  params[:member][:city],
+                                     postal:  params[:member][:postal],
+                                     address1:  params[:member][:address1],
+                                     address2:  params[:member][:address2]
+                                   }.as_json)
+      end
+
+      it 'sends the message to the AK processor' do
+        allow(ChampaignQueue).to receive(:push)
+        expect(ChampaignQueue).to receive(:push).with({
+          type: 'update_member',
+          params: {
+            akid: member.actionkit_user_id,
+            email: params[:member][:email],
+            first_name: params[:member][:first_name],
+            last_name: params[:member][:last_name],
+            country: params[:member][:country],
+            city: params[:member][:city],
+            postal: params[:member][:postal],
+            address1: params[:member][:address1],
+            address2: params[:member][:address2]
```

Comment:
What about replacing calls to params with real values? I'm a stickler for comparing against actual data. If `params[:member][:email]` returned nil (when it shouldn't), then this would still pass.